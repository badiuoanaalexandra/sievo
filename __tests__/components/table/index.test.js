import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import ProjectTable from '../../../src/components/table/index';

const mockStore = configureStore();

const initialState = {
  projectsReducer: [
      {"project":2, "description":"test", "start date": "2013-12-28T00:00:00+00:00", "savings amount":3722.41684, "currency":"NULL"},
      {"project":1, "description":"test 2", "start date": "2014-01-28T00:00:00+00:00", "savings amount":372.32, "currency":"EUR"}
  ],
  displayProjectsReducer:{changed:false, description:"", sortingOrder:"", changedProjects:null},
  uiReducer:1024
};

const store = mockStore(initialState);

describe('<ProjectTable />', () => {
  test("renders", () => {
    const wrapper = shallow(<ProjectTable store={store} />);
    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();
  });
});
