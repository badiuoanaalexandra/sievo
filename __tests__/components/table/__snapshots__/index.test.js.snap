// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectTable /> renders 1`] = `
<div>
  <div
    className="project-table"
  >
    <div>
      <div
        className="sort header-cell"
        key="header0"
      >
        <button
          className="sort "
          field="project"
          onClick={[Function]}
        >
          PROJECT
        </button>
      </div>
      <div
        className="header-cell"
        key="header1"
      >
        DESCRIPTION
      </div>
      <div
        className="header-cell"
        key="header2"
      >
        START DATE
      </div>
      <div
        className="header-cell"
        key="header3"
      >
        SAVINGS AMOUNT
      </div>
      <div
        className="header-cell"
        key="header4"
      >
        CURRENCY
      </div>
    </div>
    <div
      key="project0"
    >
      <div
        className="cell"
        key="row0"
      >
        2
      </div>
      <div
        className="cell"
        key="row1"
      >
        test
      </div>
      <div
        className="cell"
        key="row2"
      >
        2013-12-28
      </div>
      <div
        className="cell"
        key="row3"
      >
        3,722.417
      </div>
      <div
        className="cell"
        key="row4"
      />
    </div>
    <div
      key="project1"
    >
      <div
        className="cell"
        key="row0"
      >
        1
      </div>
      <div
        className="cell"
        key="row1"
      >
        test 2
      </div>
      <div
        className="cell"
        key="row2"
      >
        2014-1-28
      </div>
      <div
        className="cell"
        key="row3"
      >
        372.32
      </div>
      <div
        className="cell"
        key="row4"
      >
        EUR
      </div>
    </div>
  </div>
</div>
`;
